appcues/src/main/java/com/appcues/trait/appcues/TooltipPathExt.kt

```diff
@@ -66,7 +67,7 @@ internal sealed class TooltipPointerPosition {
      */
     abstract fun toContainerCornerRadius(cornerRadius: Dp): ContainerCornerRadius
 
-    object Top : TooltipPointerPosition() {
+    class Top(override val maxHeight: Dp? = null) : TooltipPointerPosition() {
```

Comment:
should it be a `data class` ?

Reply:
data class generates a lot of boilerplate we wont use, wont change much here, we could edit it if we want.

on the topic of changing from object to class, which led to a lot of small changes like  including "is" in front of classes on when statements, I feel like its a value that is tied to the tooltipPosition calculation, so it would make sense to have it there, I did it this way to avoid passing in an extra argument for maxHeight like we already had before. 

this way we avoid having to gather all this values in two different places, and its also an information that we may/may not need, depending on which scenario we fall into when considering the pref position and the excess and available spaces on screen

Reply:
Yeah, I think I can see how it would require handling in two different places otherwise, so that is fair. I also tested the scenario where the content is tall but there is still enough space to position the tooltip left or right, rather than shrink content, and that seems to work as expected 👍. It keeps the tall content and moves the pointer to the right side in this example, like I'd expect.

![Screenshot 2023-05-31 at 9 15 56 AM](https://github.com/appcues/appcues-android-sdk/assets/19266448/f5b003c1-c8e8-466d-a6f4-8f8b8a5a61a7)

Reply:
yea we are not changing the exiting logic, the maxHeight is calculated for when we fall into the edge cases only (two last conditions on the when statement [here](https://github.com/appcues/appcues-android-sdk/pull/409/files#diff-ade2386127bf7b29bbf36b66130e680bc4e704eddfa49fa010e1797a1983f49eR93))